packages/babel-plugin-transform-modules-amd/src/index.ts

```diff
@@ -35,7 +36,22 @@ function injectWrapper(path, wrapper) {
   amdFactory.pushContainer("body", body);
 }
 
-export default declare((api, options) => {
+export interface Options extends PluginOptions {
+  allowTopLevelThis?: boolean;
+  importInterop?: "babel" | "node";
+  loose?: boolean;
+  noInterop?: boolean;
+  strict?: boolean;
+  strictMode?: boolean;
+}
+
+export interface State {
+  requireId?: t.Identifier;
+  resolveId?: t.Identifier;
+  rejectId?: t.Identifier;
+}
+
+export default declare<State>((api, options: Options) => {
```

Comment:
A plugin can declare its own State definition, the state will be available as the second callback param of NodePath visitors.

However I have to export the State interface otherwise TS will complain at `preset-env/src/available-plugins`:

<blockquote>
TS4082: Default export of the module has or is using private name 'State'.

 74 export default {
    ~~~~~~~~~~~~~~~~
 75   "bugfix/transform-async-arrows-in-class": () => bugfixAsyncArrowsInClass,
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
... 
149   "transform-unicode-regex": () => transformUnicodeRegex,
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
150 };
    ~~
</blockquote>

I don't think the State should be exported, suggestions are welcome.

Reply:
Does TS also complain if we define it inline? i.e.

```js
export default declare<{
  requireId?: t.Identifier;
  resolveId?: t.Identifier;
  rejectId?: t.Identifier;
}>((api, options: Options) => {
```

and if we use `type` instead of `interface`?

Reply:
Using `type` works as intended, thanks!